src/components/Formik/Formik.stories.jsx

```diff
@@ -68,15 +100,55 @@ export const FormikForm = () => {
                 component={FormikTextInput}
               />
             </div>
-            <div style={{ marginBottom: '1rem' }}>
+            <div style={{ marginBottom: '1.25rem' }}>
               <Field
                 label="Email"
                 name="email"
                 id="email"
                 component={FormikTextInput}
+                isRequired
+              />
+            </div>
+            <div style={{ marginBottom: '1.25rem' }}>
+              <Field
+                label="Flavor"
+                name="flavor"
+                id="flavor"
+                options={flavorOptions}
+                component={FormikSelectInput}
+                // With a custom onChange.
+                // We preserve Formik's convention and relegate state management back to the user.
+                onChange={selectedOption => {
+                  action('change')(selectedOption);
+                  setFieldValue('flavor', selectedOption);
+                }}
+                isRequired
+              />
+            </div>
+            <div style={{ marginBottom: '1.25rem' }}>
+              <Field
+                label="Flavor no onChange"
+                name="flavor2"
+                id="flavor2"
+                options={flavorOptions}
+                component={FormikSelectInput}
+                isRequired
+              />
+            </div>
+            <div style={{ marginBottom: '1.25rem' }}>
+              <Field
+                label="Colors"
+                name="colors"
+                id="colors"
+                isMulti
+                options={colorOptions}
+                component={FormikSelectInput}
+                onChange={selectedOption => {
+                  setFieldValue('colors', selectedOption);
+                }}
```

Comment:
If the consumer manually sets the onChange to setFieldValue, then it works. However, I don't think this is our desired experience (should be like TextInput). 

So now I'm thinking we need to go back to the `generateOnChangeHandler()` for SelectInput...